File: 1248-count-number-of-nice-subarrays/1248-count-number-of-nice-subarrays.cs
public class Solution {
    public int NumberOfSubarrays(int[] nums, int k) {
        var dic = new Dictionary<int, int>(){[0]=1};
        int ret=0, curr=0;
        
        foreach(int n in nums)
        {
            curr += n % 2;
            ret += dic.GetValueOrDefault(curr-k, 0);
            dic[curr] = dic.GetValueOrDefault(curr, 0) + 1;
        }        
        return ret;
    }
}

File: 1248-count-number-of-nice-subarrays/1248-count-number-of-nice-subarrays.ts
function numberOfSubarrays(nums: number[], k: number): number {
    const map = new Map();
    map.set(0, 1);
    let ret = 0;
    let curr = 0;
    
    for(let n of nums){
        curr += n%2;
        ret += map.get(curr-k) || 0;
        map.set(curr, (map.get(curr) || 0) + 1);
    }
    return ret;
};